### PR TITLE
logging improvements to make debugging an easier experience

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Updated the logging to make useful stuff info, very useful stuff audit and debug stuff debug, some removed entirely
- Also fixed a couple of bugs in the audit log code
- Handle exceptions on posting message to Acm from celery tasks more gracefully

---
Any thoughts on the commented out trace? Remove it entirely?

### CODE REVIEW

File: app/extensions/logging/__init__.py [hunk 1]
```diff
@@ -146,10 +151,10 @@ def delete_object(cls, logger, obj, msg='', *args, **kwargs):
     @classmethod
     def patch_object(cls, logger, obj, patch_args, *args, **kwargs):
 
-        msg = ''
+        msg = 'Patch: '
         for patch in patch_args:
 
-            msg = f"{patch['op']} {patch['field_name']}"
+            msg += f"{patch['op']} {patch['field_name']}"
```
Comment (naknomum) [inline app/extensions/logging/__init__.py:157]: if there is no `value`, will this concatenation get kinda squashed together?  e.g. maybe it needs a space at front or end

**Review by naknomum — APPROVED** [review]
looks great. appreciate the cleaning of cruft and downgrading some to `debug` level.  also, love promoting some stuff to audit.

File: app/extensions/logging/__init__.py (hunk 1 above)
Comment (hwindsor) [reply app/extensions/logging/__init__.py]: Done

File: app/modules/asset_groups/models.py [hunk 2]
```diff
@@ -429,26 +433,36 @@ def send_detection_to_sage(self, model):
...
+            self.jobs = self.jobs
 
-        with db.session.begin(subtransactions=True):
-            db.session.merge(self)
+            with db.session.begin(subtransactions=True):
+                db.session.merge(self)
+        except HoustonException:
+            log.warning(
+                f'Sage Detection on AssetGroupSighting({self.guid}) Job{job_id} failed to start'
+            )
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:463]: Not a security risk, uuid and job ids are not secret

File: app/modules/asset_groups/models.py [hunk 3]
```diff
@@ -1025,7 +1039,7 @@ def create_from_metadata(cls, metadata):
         with db.session.begin(subtransactions=True):
             db.session.add(asset_group)
 
-        log.info('created asset_group %r' % asset_group)
+        log.debug('created asset_group %r' % asset_group)
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:1042]: The password printed out is the random password of the anonymous user created as part of this upload. 
But this does lead to another more serious security question:
Each request to EDM does print out the password. I suspect it shouldn't although this PR has not added this, only changed when it happens.

File: app/modules/asset_groups/models.py [hunk 4]
```diff
@@ -429,26 +433,36 @@ def send_detection_to_sage(self, model):
...
+            # we assign to one of the database attributes
+            self.jobs = self.jobs
 
-        with db.session.begin(subtransactions=True):
-            db.session.merge(self)
+            with db.session.begin(subtransactions=True):
+                db.session.merge(self)
+        except HoustonException:
+            log.warning(
+                f'Sage Detection on AssetGroupSighting({self.guid}) Job{job_id} failed to start'
```
Comment (hwindsor) [inline app/modules/asset_groups/models.py:462]: False positive uuid and job id are not secret

**Pull request merged**